apps/condo/domains/user/schema/SignInUserService.js

```diff
@@ -0,0 +1,233 @@
+        messageForUser: 'api.user.signInUser.INVALID_PHONE_NUMBER',
+    },
+    USER_NOT_FOUND: {
+        mutation: 'signInUser',
+        code: NOT_FOUND,
+        type: 'USER_NOT_FOUND',
+        message: 'User not found',
+        messageForUser: 'api.user.signInUser.USER_NOT_FOUND',
+    },
+    TOKEN_NOT_FOUND: {
+        mutation: 'signInUser',
+        variable: ['data', 'token'],
+        code: NOT_FOUND,
+        type: 'TOKEN_NOT_FOUND',
+        message: 'Token not found',
+    },
+    MULTIPLE_USERS_FOUND: {
+        mutation: 'signInUser',
+        code: NOT_FOUND,
+        type: 'MULTIPLE_USERS_FOUND',
+        message: 'Multiple users found',
+    },
+    DV_VERSION_MISMATCH: {
+        query: 'signInUser',
+        variable: ['data', 'dv'],
+        code: BAD_USER_INPUT,
+        type: DV_VERSION_MISMATCH,
+        message: 'Wrong value for data version number',
+    },
+    WRONG_SENDER_FORMAT: {
+        query: 'signInUser',
+        variable: ['data', 'sender'],
+        code: BAD_USER_INPUT,
+        type: WRONG_FORMAT,
+        message: 'Invalid format of "sender" field value',
+        correctExample: '{ dv: 1, fingerprint: \'example-fingerprint-alphanumeric-value\'}',
+    },
+}
+
+
+const REQUIRED_USER_FIELDS_BY_TYPE = {
+    [RESIDENT]: ['phone'],
+    [STAFF]: ['phone', 'name', 'password'],
+}
+
+const getUserPayload = (user, userData) => {
+    const payload = {}
+    for (const [key, value] of Object.entries(userData))  {
+        if (!value) continue
+        if (get(user, key)) continue
+        payload[key] = value
+    }
+    return payload
+}
+
+const checkRequiredUserFields = (requiredFields, userData) => {
+    if (!Array.isArray(requiredFields)) throw new Error('"requiredFields" should be an array')
+
+    const missingFields = []
+    for (const field of requiredFields) {
+        const value = get(userData, field, null)
+        if (!value && typeof value !== 'boolean' && typeof value !== 'number') missingFields.push(field)
+    }
+    return { missingRequiredFields: missingFields.length > 0, missingFields }
+}
+
+const SignInUserService = new GQLCustomSchema('SignInUserService', {
+    types: [
+        {
+            access: true,
+            type: 'input SignInUserUserDataInput { phone: String, name: String, email: String, password: String, meta: JSON }',
+        },
+        {
+            access: true,
+            type: 'input SignInUserInput { dv: Int!, sender: JSON!, confirmActionToken: String!, userType: UserTypeType!, userData: SignInUserUserDataInput }',
+        },
+        {
+            access: true,
+            type: 'type SignInUserOutput { user: User, token: String! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canSignInUser,
+            schema: 'signInUser(data: SignInUserInput!): SignInUserOutput',
+            doc: {
+                summary: 'This mutation authorizes the user without a password, after confirming the phone number.\n\n' +
```

Comment:
```suggestion
                summary: 'This mutation authorizes the user by token, after confirming the phone number.\n\n' +
```

Reply:
Why separate token is used, why not just ConfirmAction id, since both are uuids....